dima/react-movies/src/components/searchBar/searchBar.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import SearchFilter from "./searchFilter";
+import "./searchBar.css";
+import ErrorBoundary from "../errorBoundary";
+import * as QueryString from "query-string";
+import FilterProperty from "../../enums/FilterPropery";
+import { RouteComponentProps } from "react-router";
+import ParamsToPush from "../../interfaces/paramsToPush";
+interface SearchBarProps {
+  location: RouteComponentProps["location"];
+  pushParamsOnSubmit: (urlParams: ParamsToPush) => void;
+}
+
+interface SearchBarState {
+  searchBarValue: string | undefined;
+  searchBy: FilterProperty;
+}
+
+export default class SearchBar extends React.Component<
+  SearchBarProps,
+  SearchBarState
+> {
+  state: SearchBarState = {
+    searchBarValue: "",
+    searchBy: FilterProperty.title,
+  };
+
+  componentDidMount() {
+    if (QueryString.parse(this.props.location.search).searchBy) {
+      this.toggleSearchCategoryByURL();
+    }
+    const { search } = QueryString.parse(this.props.location.search);
+    if (search) {
+      this.setState({ searchBarValue: search.toString() });
+    }
+  }
+  componentDidUpdate(prevProps: SearchBarProps) {
+    if (
+      QueryString.parse(prevProps.location.search).search !==
+      QueryString.parse(this.props.location.search).search
+    ) {
+      const { search } = QueryString.parse(this.props.location.search);
+      if (search) {
+        this.setState({ searchBarValue: search.toString() });
+      } else {
+        this.setState({ searchBarValue: "" });
+      }
```

Comment:
You have to update searchBy as well. Haven't noticed an issue here on the demo. 
You can call your toggleSearchCategoryByURL(setFiltersFromUrl) func here